de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/IActivityQueuer.java

```diff
@@ -0,0 +1,22 @@
+package de.fu_berlin.inf.dpp.session;
+
+import java.util.List;
+
+import de.fu_berlin.inf.dpp.activities.IActivity;
+import de.fu_berlin.inf.dpp.activities.IResourceActivity;
+
+public interface IActivityQueuer {
+
+    /**
+     * Processes the incoming {@linkplain IActivity activities} and decides
+     * which {@linkplain IResourceActivity resource related activities} should
+     * be queued. The method returns all other activities.
+     * <p>
+     * If flushing of the queue was previously requested, than this method will
+     * additionally return activities from the queue.
+     * 
+     * @param activities
+     * @return activities that are not queued
+     */
+    public List<IActivity> process(final List<IActivity> activities);
```

Comment:
On the other hand, the addition of another "ActivityQueuer" implementation makes it seem like maybe there should be a general IActivityQueuer implementation.

Reply:
Yes, another Impl is the reason for this and the process method is the only common for both.
To avoid the naming confusion I renamed the ActivityQueuer in #155 and In #156 I add an resource based queuing mechanism, which hopefully explains the reason more practical.